### DISCUSSION TITLE
Need to change id_order to GUID/UUID

### DISCUSSION THREAD

**Discussion by RomSunZ** [PrestaShop/PrestaShop#40188] [Ideas]
Please replace id_order in every links (customer history and so on) to GUID/UUID for sequrity reasons. Customer must not be shown easily guessable autoincreament integers.

---
#### Is your feature request related to a problem?
Customer must not be shown easily guessable autoincreament integers.

In the FO => Home / Your account / Order history / Order details
The URL displayed http://xxxxx/en/index.php?controller=order-detail&id_order=5 ( Friendly URL is enabled) ( id_order is displayed in the URL)

#### Describe the solution you'd like
Replace id_order in every links (customer history and so on) to GUID/UUID for sequrity reasons.

I think that every FO page where are links to any order should be replaced to GUID, not only Order history / Order details, but for example checkout page for order confirmation. This is more complex problem than just order history page.

#### Additional context

![image](https://user-images.githubusercontent.com/16067358/47416000-07fd8200-d775-11e8-8da7-ae04d95b63aa.png)

**Comment by khouloudbelguith** [comment]
Hi @RomSunZ,

In the FO => Home / Your account / Order history / Order details
The URL displayed http://xxxxx/en/index.php?controller=order-detail&id_order=5 ( Friendly URL is enabled) ( id_order is displayed in the URL)
![image](https://user-images.githubusercontent.com/16067358/47416000-07fd8200-d775-11e8-8da7-ae04d95b63aa.png)
In the 1.6, Order details is displayed at the bottom of the page, under the list of orders.
![image](https://user-images.githubusercontent.com/16067358/47416164-5f9bed80-d775-11e8-9e3c-73f46ac16416.png)

@marionf what do you think?
Thanks!

**Comment by RomSunZ** [comment]
I think that every FO page where are links to any order should be replaced to GUID, not only Order history / Order details, but for example checkout page for order confirmation. This is more complex problem than just order history page.

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team friendly reminder :wink: What is the status of this issue?

**Comment by matks** [comment]
Hi,

1. This feature request is very good. Indeed using SQL incrementable integers is neither secure nor a good thing, especially for something as important as Order IDs. By the way there is already an available replacement which is the Order reference, but as mentioned by @RomSunZ it would be needed to use them everywhere.

2. However doing this change would be a massive BC breaks as it would change
- the SQL data structure of a widely used table and entity
- all the code around, that rely on this identifier
- the URLs, the FO, the BO...

So we wont be able to do it until a new major version, PrestaShop 8 or PrestaShop 9. Doing this change in a minor version would create a huge breaking change impacting existing modules and themes.

**Comment by hibatallahAouadni** [comment]
@matks you mean PrestaShop 1.8 or PrestaShop 1.9, is that so? :sweat_smile:

**Comment by matks** [comment]
@hibatallahAouadni We plan to stop using the 1.X notation from next major. The `1.` means nothing.

It's not 100% sure but hopefully, after PS 1.7.9 will come PS 8.0.0
